platform/jvm/MacAppNotarizer.kt

```diff
@@ -1,129 +1,51 @@
-import com.vaticle.bazel.distribution.platform.jvm.MacAppNotarizer.NotarizationInfoResult.Status.PENDING
-import com.vaticle.bazel.distribution.platform.jvm.MacAppNotarizer.NotarizationInfoResult.Status.REJECTED
-import com.vaticle.bazel.distribution.platform.jvm.MacAppNotarizer.StatusPoller.LOG_FILE_URL
-import com.vaticle.bazel.distribution.platform.jvm.MacAppNotarizer.StatusPoller.MAX_RETRIES
-import com.vaticle.bazel.distribution.platform.jvm.MacAppNotarizer.StatusPoller.POLL_INTERVAL_MS
-import com.vaticle.bazel.distribution.platform.jvm.MacAppNotarizer.StatusPoller.STATUS_MESSAGE_PACKAGE_APPROVED
+import com.vaticle.bazel.distribution.platform.jvm.MacAppNotarizer.Args.SUBMIT
+import com.vaticle.bazel.distribution.platform.jvm.MacAppNotarizer.Args.TEAM_ID
+import com.vaticle.bazel.distribution.platform.jvm.MacAppNotarizer.Args.TIMEOUT
+import com.vaticle.bazel.distribution.platform.jvm.MacAppNotarizer.Args.WAIT
 import com.vaticle.bazel.distribution.platform.jvm.ShellArgs.Programs.XCRUN
 import java.nio.file.Path
 
-class MacAppNotarizer(private val options: Options.AppleCodeSigning, private val dmgFilename: String, private val dmgPath: Path) {
-    fun notarize() {
-        val requestUUID = parseNotarizeResult(shell.execute(notarizeCommand()).outputString())
-        logger.debug { "Notarization request UUID: $requestUUID" }
-        waitForPackageApproval(requestUUID)
+class MacAppNotarizer(private val dmgPath: Path) {
+    fun notarize(appleCodeSigning: Options.AppleCodeSigning) {
+        shell.execute(notarizeCommand(appleCodeSigning)).outputString()
         markPackageAsApproved()
     }
 
-    private fun notarizeCommand(): Shell.Command {
-        // TODO: xcrun altool --notarize-app is deprecated in Xcode 13: see
-        //       https://developer.apple.com/documentation/security/notarizing_macos_software_before_distribution/customizing_the_notarization_workflow?preferredLanguage=occ
+    private fun notarizeCommand(appleCodeSigning: Options.AppleCodeSigning): Shell.Command {
         return Shell.Command(
-            Shell.Command.arg(XCRUN), Shell.Command.arg(ALTOOL), Shell.Command.arg(NOTARIZE_APP),
-            Shell.Command.arg(PRIMARY_BUNDLE_ID), Shell.Command.arg(options.macAppID),
-            Shell.Command.arg(USERNAME), Shell.Command.arg(options.appleID, printable = false),
-            Shell.Command.arg(PASSWORD), Shell.Command.arg(options.appleIDPassword, printable = false),
-            Shell.Command.arg(FILE), Shell.Command.arg(dmgPath.toString())
-        )
-    }
-
-    private fun parseNotarizeResult(value: String): String {
-        return Regex("RequestUUID = ([a-z0-9\\-]{36})").find(value)?.groupValues?.get(1)
-            ?: throw IllegalStateException("Notarization failed: the response $value from " +
-                    "'xcrun altool --notarize-app' does not contain a valid RequestUUID")
-    }
-
-    private fun waitForPackageApproval(requestUUID: String) {
```

Comment:
We no longer need to do this as `notarytool` provides a handy `--wait` flag for us to use (along with a timeout).